### PR TITLE
feat: Improve get total data by day

### PR DESCRIPTION
**Pull request by hibitness-an-nguyen2**

### CODE REVIEW

File: packages/health/ios/Classes/SwiftHealthPlugin.swift [hunk 1]
```diff
@@ -719,6 +724,82 @@ public class SwiftHealthPlugin: NSObject, FlutterPlugin {
...
+  func getTotalDataTypeByDayInInterval(call: FlutterMethodCall, result: @escaping FlutterResult) {
+    guard let arguments = call.arguments as? NSDictionary,
+        let dataTypeKey = (arguments["dataTypeKey"] as? String),
+        let dataUnitKey = (arguments["dataUnitKey"] as? String),
+        let startTime = (arguments["startTime"] as? NSNumber),
+        let endTime = (arguments["endTime"] as? NSNumber)
+    else {
+        result(
+            FlutterError(
+                code: "INVALID ARGUMENTS",
```
Comment (hibitness-quan-nguyen2) [inline packages/health/ios/Classes/SwiftHealthPlugin.swift]:
```suggestion
                code: "INVALID_ARGUMENTS",
```

File: packages/health/ios/Classes/SwiftHealthPlugin.swift [hunk 2]
```diff
@@ -719,6 +724,82 @@ public class SwiftHealthPlugin: NSObject, FlutterPlugin {
...
+    // Convert dates from milliseconds to Date()
+    let dateFrom = Date(timeIntervalSince1970: startTime.doubleValue / 1000)
+    let dateTo = Date(timeIntervalSince1970: endTime.doubleValue / 1000)
+    
+    guard let unit = unitDict[dataUnitKey],
+        let dataType = dataTypeLookUp(key: dataTypeKey) as? HKQuantityType
+    else {
+        result(
+            FlutterError(
+                code: "INVALID INPUT TYPE",
```
Comment (hibitness-quan-nguyen2) [inline packages/health/ios/Classes/SwiftHealthPlugin.swift]:
```suggestion
                code: "INVALID_INPUT_TYPE",
```

File: packages/health/lib/src/health_factory.dart [hunk 3]
```diff
@@ -834,4 +834,85 @@ class HealthFactory {
...
+        throw HealthException(
+          dataType,
+          'Not available on platform $_platformType',
+        );
+      }
+    });
+
+    _deviceId ??= _platformType == PlatformType.ANDROID
+        ? (await _deviceInfo.androidInfo).id
+        : (await _deviceInfo.iosInfo).identifierForVendor;
```
Comment (hibitness-quan-nguyen2) [inline packages/health/lib/src/health_factory.dart:871]:
```suggestion
    _deviceId ??= (await _deviceInfo.iosInfo).identifierForVendor;
```

File: packages/health/ios/Classes/SwiftHealthPlugin.swift [hunk 4]
```diff
@@ -719,6 +724,82 @@ public class SwiftHealthPlugin: NSObject, FlutterPlugin {
...
+                message: "Invalid data type or unit type.",
+                details: nil
+            )
+        )
+        return
+    }
+
+    let predicate = HKQuery.predicateForSamples(
+        withStart: dateFrom, end: dateTo, options: .strictStartDate
+    )
```
Comment (hibitness-quan-nguyen2) [inline packages/health/ios/Classes/SwiftHealthPlugin.swift]: Remove if unused

**Review by hibitness-ichikawa-masaaki — APPROVED** [review]
LGTM!

**Pull request merged**